> Mon .env
> 
> ```
> DATABASE_URL=postgresql://postgres@localhost/pix
> TEST_DATABASE_URL=postgresql://postgres@localhost/pix_test
> ```
> 
> J'ai supprimé les conteneurs et lancé le script.. et c'est KO 😢
> `npm run configure`
> 
> ```
> > pix-api@3.93.0 db:migrate /home/topi/Documents/OCTO/Missions/Pix/code/repo/pix/api
> > knex --knexfile db/knexfile.js migrate:latest
> 
> Working directory changed to ~/Documents/OCTO/Missions/Pix/code/repo/pix/api/db
> Using environment: development
> error: database "pix" does not exist
>     at Function.get (/home/topi/Documents/OCTO/Missions/Pix/code/repo/pix/api/node_modules/knex/lib/knex-builder/make-knex.js:43:26)
>     at getSchemaBuilder (/home/topi/Documents/OCTO/Missions/Pix/code/repo/pix/api/node_modules/knex/lib/migrations/migrate/table-creator.js:61:17)
>     at ensureTable (/home/topi/Documents/OCTO/Missions/Pix/code/repo/pix/api/node_modules/knex/lib/migrations/migrate/table-creator.js:11:10)
>     at listCompleted (/home/topi/Documents/OCTO/Missions/Pix/code/repo/pix/api/node_modules/knex/lib/migrations/migrate/migration-list-resolver.js:12:9)
>     at Object.listAllAndCompleted (/home/topi/Documents/OCTO/Missions/Pix/code/repo/pix/api/node_modules/knex/lib/migrations/migrate/migration-list-resolver.js:28:5)
>     at Migrator.latest (/home/topi/Documents/OCTO/Missions/Pix/code/repo/pix/api/node_modules/knex/lib/migrations/migrate/Migrator.js:64:57)
>     at Command.<anonymous> (/home/topi/Documents/OCTO/Missions/Pix/code/repo/pix/api/node_modules/knex/bin/cli.js:232:55)
> npm ERR! code ELIFECYCLE
> ```

Le `db:prepare` fait un drop de la db avant d'en créer une nouvelle du coup lors du premier lancement du projet il ne trouve de db

--- a/api/scripts/PgClient.js
+++ b/api/scripts/PgClient.js
@@ -2,12 +2,21 @@ const { Client } = require('pg');
 
 class PgClient {
   constructor(databaseUrl) {
-    this.client = new Client(databaseUrl);
-    this.client.connect();
+    this.client = new Client({ connectionString: databaseUrl, connectionTimeoutMillis: 10000 });
+  }
+
+  static async getClient(databaseUrl) {
+    const instance = new PgClient(databaseUrl);
+    try {
+      await instance.client.connect();
+    } catch (error) {
+      console.error('Database error', error);
+    }
+    return instance;
   }
 
   end() {
-    this.client.end();
+    return this.client.end();
   }
 
   query_and_log(query) {

--- a/api/scripts/database/create-database.js
+++ b/api/scripts/database/create-database.js
@@ -10,20 +10,18 @@ const DB_TO_CREATE_NAME = url.pathname.slice(1);
 
 url.pathname = '/postgres';
 
-const client = new PgClient(url.href);
-
-client.query_and_log(`CREATE DATABASE ${DB_TO_CREATE_NAME};`)
-  .then(function() {
+PgClient.getClient(url.href).then(async (client) => {
+  try {
+    await client.query_and_log(`CREATE DATABASE ${DB_TO_CREATE_NAME};`);
     console.log('Database created');
-    client.end();
+    await client.end();
     process.exit(0);
-  }).catch((error) => {
+  } catch (error) {
     if (error.code === PGSQL_DUPLICATE_DATABASE_ERROR) {
       console.log(`Database ${DB_TO_CREATE_NAME} already created`);
-      client.end();
+      await client.end();
       process.exit(0);
     }
-    else {
-      throw error;
-    }
-  });
+  }
+});
+

--- a/api/scripts/database/drop-database.js
+++ b/api/scripts/database/drop-database.js
@@ -10,20 +10,18 @@ const DB_TO_DELETE_NAME = url.pathname.slice(1);
 
 url.pathname = '/postgres';
 
-const client = new PgClient(url.href);
-
-client.query_and_log(`DROP DATABASE ${DB_TO_DELETE_NAME};`)
-  .then(() => {
+PgClient.getClient(url.href).then(async (client) => {
+  try {
+    await client.query_and_log(`DROP DATABASE ${DB_TO_DELETE_NAME};`);
     console.log('Database dropped');
-    client.end();
+    await client.end();
     process.exit(0);
-  }).catch((error) => {
+  } catch (error) {
     if (error.code === PGSQL_NON_EXISTENT_DATABASE_ERROR) {
       console.log(`Database ${DB_TO_DELETE_NAME} does not exist`);
-      client.end();
+      await client.end();
       process.exit(0);
     }
-    else {
-      throw error;
-    }
-  });
+  }
+});
+

--- a/api/scripts/delete-assessment.js
+++ b/api/scripts/delete-assessment.js
@@ -2,20 +2,20 @@
 /* eslint no-console: ["off"] */
 const PgClient = require('./PgClient');
 
-function initialize() {
-  const client = new PgClient(process.env.DATABASE_URL);
+async function initialize() {
+  const client = await PgClient.getClient(process.env.DATABASE_URL);
 
   const assessment_id = process.argv[2];
   return { client, assessment_id };
 }
 
-function terminate(client) {
-  client.end();
+async function terminate(client) {
+  await client.end();
   console.log('END');
 }
 
-function main() {
-  const { client, assessment_id } = initialize();
+async function main() {
+  const { client, assessment_id } = await initialize();
 
   const queryBuilder = new ScriptQueryBuilder();
   const userEraser = new AssessmentEraser(client, queryBuilder, assessment_id);
@@ -54,9 +54,9 @@ class AssessmentEraser {
         this.queryBuilder.delete_competence_marks_from_assessment_ids(this.assessment_id),
       ])
       .then((queries) => Promise.all(
-        queries.map((query) => {
-          this.client.query_and_log(query);
-        }),
+        queries.map((query) =>
+          this.client.query_and_log(query),
+        ),
       ))
       .then(() => this.queryBuilder.delete_assessment_results_from_assessment_ids(this.assessment_id))
       .then((query) => this.client.query_and_log(query));

--- a/api/scripts/delete-user.js
+++ b/api/scripts/delete-user.js
@@ -2,20 +2,20 @@
 /* eslint no-console: ["off"] */
 const PgClient = require('./PgClient');
 
-function initialize() {
-  const client = new PgClient(process.env.DATABASE_URL);
+async function initialize() {
+  const client = await PgClient.getClient(process.env.DATABASE_URL);
 
   const user_email = process.argv[2];
   return { client, user_email };
 }
 
-function terminate(client) {
-  client.end();
+async function terminate(client) {
+  await client.end();
   console.log('END');
 }
 
-function main() {
-  const { client, user_email } = initialize();
+async function main() {
+  const { client, user_email } = await initialize();
   const queryBuilder = new ScriptQueryBuilder();
   const clientQueryAdapter = new ClientQueryAdapter();
   const userEraser = new UserEraser(client, queryBuilder, clientQueryAdapter);

--- a/scripts/configure.sh
+++ b/scripts/configure.sh
@@ -6,6 +6,8 @@ set -o pipefail
 EXPECTED_NODE_VERSION="v14.17.0"
 EXPECTED_NPM_VERSION="6.14.13"
 
+function timeout() { perl -e 'alarm shift; exec @ARGV' "$@"; }
+
 function display_banner() {
   echo "                                                    "
   echo "                                                    "
@@ -95,19 +97,24 @@ function install_apps_dependencies() {
 function setup_and_run_infrastructure() {
   echo "Starting infrastructure building blocks…"
 
-  docker-compose up -d
-  (cd api && npm run db:migrate)
+  docker-compose up -d --force-recreate
 
   echo "✅ PostgreSQL and Redis servers started (using Docker Compose)."
   echo ""
-}
 
-function load_seed() {
-  echo "Loading seed data"
+  echo "Waiting for PostgreSQL server to be ready…"
 
-  (cd api && npm run db:seed)
+  timeout 20s bash -c "until docker exec pix_postgres_1 pg_isready ; do sleep 1 ; done"
 
-  echo "✅ Seed data loaded"
+  echo "✅ PostgreSQL server is ready."
+  echo ""
+
+  echo "Creating database"
+
+  # It drops and creates database then load the seed.
+  (cd api && npm run db:reset)
+
+  echo "✅ Database created"
   echo ""
 }
 
@@ -147,6 +154,5 @@ verify_prerequesite_programs
 generate_environment_config_file
 install_apps_dependencies
 setup_and_run_infrastructure
-load_seed
 execute_apps_tests
 display_footer

